numpy/ma/core.py

```diff
@@ -3305,6 +3305,10 @@ def _scalar_heuristic(arr, elem):
                     # which should have shape () to avoid breaking several
                     # methods. There is no great way out, so set to
                     # first element.  See issue #6723.
+                    # something like gh-15895 has happened if this check fails.
+                    # _fill_value should always be an ndarray.
+                    if not isinstance(dout._fill_value, np.ndarray):
+                        raise RuntimeError('Internal NumPy error.')
```

Comment:
nit: can you move these lines up to line 3299? Where you've put them has separated the comment above from the code it describes.